src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/Delivery/ClusterShardingDeliveryGracefulShutdownSpec.cs

```diff
@@ -0,0 +1,292 @@
+                        _probe.Tell(CoordinatorTerminated.Instance);
+                        return true;
+
+                    case Terminated t when t.ActorRef.Equals(_region):
+                        _probe.Tell(RegionTerminated.Instance);
+                        return true;
+                }
+                return false;
+            }
+        }
+
+        private sealed class MessageExtractor: IMessageExtractor
+        {
+            public string EntityId(object message)
+                => message switch
+                {
+                    SlowStopConsumerEntity.Job j => j.Payload.ToString(),
+                    _ => null
+                };
+
+            public object EntityMessage(object message)
+                => message;
+
+            public string ShardId(object message)
+                => message switch
+                {
+                    SlowStopConsumerEntity.Job j => j.Payload.ToString(),
+                    _ => null
+                };
+
+            public string ShardId(string entityId, object messageHint = null)
+                => entityId;
+        }
+
+        private const string TypeName = "SlowStopEntity";
+        private IActorRef _producer;
+        private IActorRef _producerController;
+
+        protected ClusterShardingDeliveryGracefulShutdownSpec(ClusterShardingDeliveryGracefulShutdownSpecConfig config, Type type)
+            : base(config, type)
+        {
+        }
+
+        private IActorRef CreateProducer(string producerId)
+        {
+            _producerController =
+                Sys.ActorOf(
+                    ShardingProducerController.Create<SlowStopConsumerEntity.Job>(
+                        producerId: producerId, 
+                        shardRegion: ClusterSharding.Get(Sys).ShardRegion(TypeName), 
+                        durableQueue: Option<Props>.None,
+                        settings: ShardingProducerController.Settings.Create(Sys)), 
+                    "shardingProducerController");
+            _producer = Sys.ActorOf(Props.Create(() => new TestShardingProducer(_producerController, TestActor)),
+                "producer");
+            return _producer;
+        }
+        
+        private IActorRef StartSharding()
+        {
+            return ClusterSharding.Get(Sys).Start(
+                typeName: TypeName, 
+                entityPropsFactory: e => ShardingConsumerController.Create<SlowStopConsumerEntity.Job>(
+                    c => Props.Create(() => new SlowStopConsumerEntity(e, c)),
+                    ShardingConsumerController.Settings.Create(Sys)), 
+                settings: Settings.Value.WithRole(null),
+                messageExtractor: new MessageExtractor(),
+                allocationStrategy: ShardAllocationStrategy.LeastShardAllocationStrategy(absoluteLimit: 2, relativeLimit: 1.0),
+                handOffStopMessage: SlowStopConsumerEntity.Stop.Instance);
```

Comment:
This would not work without the new `allow-bypass` feature

Reply:
LGTM